nomos-blend/message/src/mock/mod.rs

```diff
@@ -60,21 +60,24 @@ impl BlendMessage for MockBlendMessage {
     fn unwrap(
         message: &[u8],
         private_key: &Self::PrivateKey,
-    ) -> Result<(Vec<u8>, bool), Self::Error> {
+    ) -> Result<(Vec<u8>, bool), MessageUnwrapError<Self::Error>> {
         if message.len() != MESSAGE_SIZE {
-            return Err(Error::InvalidBlendMessage);
+            return Err(MessageUnwrapError::Other(Error::InvalidBlendMessage));
```

Comment:
I guess we could also provide a `From` implementation for any type that implements `Error`, so we could do `return Err(Error::InvalidBlendMessage.into())` here? Does not really matter, but it might improve ergonomics a bit. And we could also use `?` which performs `From` conversion automatically.